src/config.js

```diff
@@ -257,13 +257,20 @@ export default class Config {
    */
 
   getOfflineMirrorPath(tarUrl: ?string): ?string {
-    const registry = this.registries.npm;
-    if (registry == null) {
-      return null;
+    //
+    let mirrorPath;
+
+    for (const key of Object.keys(this.registries)) {
+      const registry = this.registries[key];
+      const registryMirrorPath = registry.config['yarn-offline-mirror'];
+
+      if (registryMirrorPath == null) {
+        continue;
```

Comment:
Sure.
The check was not needed when you cycle over this.registries entries but if you replace the cycle with this.registries.npm and this.registries.yarn then you need to check for null